packages/expo-notifications/ios/EXNotifications/Permissions/PermissionsModule.swift

```diff
@@ -0,0 +1,81 @@
+//  Copyright © 2024 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import UIKit
+import MachO
+
+public class PermissionsModule: Module {
+  var permissionsManager: (any EXPermissionsInterface)?
+  var requester: EXUserFacingNotificationsPermissionsRequester =
+    EXUserFacingNotificationsPermissionsRequester()
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoNotificationPermissionsModule")
+
+    OnCreate {
+      appContext?.permissions?.register([
+        requester
+      ])
+    }
+
+    AsyncFunction("getPermissionsAsync") { (promise: Promise) in
+      appContext?
```

Comment:
if `appContext?.permissions` is nil, then the method will never resolve, will it?

Seems good to reject the promise if that's the case.